sanic/request.py

```diff
@@ -172,6 +172,10 @@ async def respond(
         headers: Optional[Union[Header, Dict[str, str]]] = None,
         content_type: Optional[str] = None,
     ):
+        if isinstance(self.stream, Http) and self.stream.stage is Stage.IDLE:
+            raise SanicException(
+                "Another response was sent previously."
+            )
```

Comment:
What is `self.stream` otherwise? `None`?

I would either check for `is None` or:

```python
try:
    if self.stream.stage is Stage.IDLE:
        raise SanicException(...)
except AttributeError:
    ...
```

Reply:
https://github.com/sanic-org/sanic-testing/issues/27

In some test cases, the stream is not `Http` 😂

Reply:
Got it. That brings up the earlier point. I think the more proper way to handle this then would be to add stage to that as discussed elsewhere.

Reply:
Can you remind me what's the proper way? I don't remember I saw it somewhere else...